assets/js/modules/analytics/datastore/accounts.js

```diff
@@ -80,12 +84,105 @@ export const actions = {
 			type: RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_FAILED,
 		};
 	},
+
+	/**
+	 * Creates a new Analytics account.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} args              Argument params.
+	 * @param {string} args.accountName  Google Analytics account name.
+	 * @param {string} args.propertyName Google Analytics property name.
+	 * @param {string} args.profileName  Google Analytics profile name.
+	 * @param {string} args.timezone     Google Analytics timezone.
+	 * @return {Function} Generator function action.
+	 */
+	*createAccount( { accountName, propertyName, profileName, timezone } ) {
+		invariant( accountName, 'accountName is required.' );
+		invariant( propertyName, 'propertyName is required.' );
+		invariant( profileName, 'profileName is required.' );
+		invariant( timezone, 'timezone is required.' );
+
+		try {
+			const accountTicket = yield actions.fetchCreateAccount( { accountName, propertyName, profileName, timezone } );
+			yield {
+				payload: {},
+				type: CREATE_ACCOUNT_FINISHED,
+			};
+
+			return actions.receiveCreateAccount( { accountTicket } );
+		} catch ( error ) {
+			// TODO: Implement an error handler store or some kind of centralized
+			// place for error dispatch...
+			return actions.receiveCreateAccountFailed( { accountName, error } );
+		}
+	},
+
+	*fetchCreateAccount( { accountName, propertyName, profileName, timezone } ) {
+		yield {
+			payload: {},
+			type: CREATE_ACCOUNT_STARTED,
+		};
+
+		return {
+			payload: { accountName, propertyName, profileName, timezone },
+			type: FETCH_CREATE_ACCOUNT,
+		};
+	},
+
+	/**
+	 * Adds a account ticket to process.
+	 *
+	 * Adds the newly-created account ticket to the data store.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} args               Argument params.
+	 * @param {Object} args.accountTicket Google Analytics create account ticket object.
+	 */
+	receiveCreateAccount( { accountTicket } ) {
+		invariant( accountTicket, 'accountTicket is required.' );
+
+		// Once we have an account ticket, redirect the user to accept the Terms of Service.
+		const { id } = accountTicket;
+		if ( id ) {
+			/* eslint-disable no-restricted-globals */
+			window.location.assign( `https://analytics.google.com/analytics/web/?provisioningSignup=false#management/TermsOfService/?api.accountTicketId=${ id }` );
+			/* eslint-enable no-restricted-globals */
+		}
+	},
+
+	/**
+	 * Logs an error with account creation.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} args       Argument params.
+	 * @param {Object} args.error Error object.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveCreateAccountFailed( { error } ) {
+		return {
+			payload: { error },
+			type: RECEIVE_CREATE_ACCOUNT_FAILED,
+		};
+	},
```

Comment:
Per the new conventions, we don't need a dedicated action creator for the fail (now called `CATCH_`) case as this is handled in the main action's generator.

Reply:
@aaemnnosttv Can you review what I have now and let me know if that looks right?